src/lib/io/include/sol-uart.h

```diff
@@ -238,7 +251,18 @@ void sol_uart_close(struct sol_uart *uart);
  * @note Caller should guarantee that tx buffer will not be freed until
  * callback is called.
  */
-bool sol_uart_write(struct sol_uart *uart, const uint8_t *tx, unsigned int length, void (*tx_cb)(void *data, struct sol_uart *uart, uint8_t *tx, int status), const void *data);
+int sol_uart_write(struct sol_uart *uart, const uint8_t *tx, size_t length, void (*tx_cb)(void *data, struct sol_uart *uart, uint8_t *tx, int status), const void *data);
```

Comment:
ah, it is worth to mention we're trying to have the pattern to fulfill the requirement: fast data producers must have a way to know the pending data is piling up so they can stop producing data.

Like "cat /dev/zero | slow-reader", once the pipe is filled, the `cat` process will be stuck into blocking `write()`. We can't just block in soletta due the main loop, so at least a way to notify data wasn't fully sent must be available.